drivers/sensor/mcp9600/mcp9600.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022, Maxmillion McLaughlin
+ * Copyright (c) 2020, SER Consulting LLC / Steven Riedl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_mcp9600
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(MCP9600, CONFIG_SENSOR_LOG_LEVEL);
+
+#define MCP9600_REG_TEMP_HOT 0x00
+
+#define MCP9600_REG_TEMP_DIFF 0x01
+#define MCP9600_REG_TEMP_COLD 0x02
+#define MCP9600_REG_RAW_ADC   0x03
+
+#define MCP9600_REG_STATUS 0x04
+
+#define MCP9600_REG_TC_CONFIG  0x05
+#define MCP9600_REG_DEV_CONFIG 0x06
+#define MCP9600_REG_A1_CONFIG  0x07
+
+#define MCP9600_REG_A2_CONFIG 0x08
+#define MCP9600_REG_A3_CONFIG 0x09
+#define MCP9600_REG_A4_CONFIG 0x0A
+#define MCP9600_A1_HYST	      0x0B
+
+#define MCP9600_A2_HYST	 0x0C
+#define MCP9600_A3_HYST	 0x0D
+#define MCP9600_A4_HYST	 0x0E
+#define MCP9600_A1_LIMIT 0x0F
+
+#define MCP9600_A2_LIMIT	0x10
+#define MCP9600_A3_LIMIT	0x11
+#define MCP9600_A4_LIMIT	0x12
+#define MCP9600_REG_ID_REVISION 0x13
+
+struct mcp9600_data {
+	int32_t temp;
+};
+
+struct mcp9600_config {
+	const struct i2c_dt_spec bus;
+};
+
+static int mcp9600_reg_read(const struct device *dev, uint8_t start, uint8_t *buf, int size)
+{
+	const struct mcp9600_config *cfg = dev->config;
+
+	return i2c_burst_read_dt(&cfg->bus, start, buf, size);
+}
+
+static int mcp9600_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct mcp9600_data *data = dev->data;
+	uint8_t buf[2];
+	int ret;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_AMBIENT_TEMP);
+
+	/* read signed 16 bit double-buffered register value */
+	ret = mcp9600_reg_read(dev, MCP9600_REG_TEMP_HOT, buf, sizeof(buf));
+	if (ret < 0) {
+		data->temp = 1;
+		return ret;
+	}
+
+	/* device's hot junction register is a signed int */
+	data->temp = (int32_t)(int16_t)(buf[0] << 8) | buf[1];
+
+	/* 0.0625C resolution per LSB */
+	data->temp *= 62500;
```

Comment:
This constant could be places into this sensor binding allowing to be set per-application.

Reply:
I don't love having magic numbers floating around either. I'd have to add support for modifying the ADC resolution though - maybe a future feature